### PR TITLE
refactor: webpack SharedWorker

### PR DESCRIPTION
**Pull request by jiechud**
This PR contains a:

- [ ] **bugfix**
- [x ] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case
no

### Breaking Changes
no

### Additional Info

add isLoaderWorker options

### CODE REVIEW

File: src/workers/InlineWorker.js [hunk 1]
```diff
@@ -24,15 +32,15 @@ module.exports = function (content, url) {
         blob = new Blob([content]);
       }
 
-      return new Worker(URL.createObjectURL(blob));
+      return new createWorker(URL.createObjectURL(blob), options.isSharedWorker);
```
Comment (A) [inline src/workers/InlineWorker.js]: Mistype?

Comment (A) [reply src/workers/InlineWorker.js]: Confusing to see both `new` and `CreateWorker()`, that not looks like a constructor

File: src/workers/InlineWorker.js [hunk 2]
```diff
@@ -2,7 +2,15 @@
 
 var URL = window.URL || window.webkitURL;
 
-module.exports = function (content, url) {
+function CreateWorker(url, isSharedWorker) {
+  if (isSharedWorker) {
+    return new isSharedWorker(url);
```
Comment (developit) [inline src/workers/InlineWorker.js]:
```suggestion
    return new SharedWorker(url);
```

**Comment by evilebottnawi** [comment]
Done here https://github.com/webpack-contrib/worker-loader/pull/247

**Pull request closed without merging**